machine.go

```diff
@@ -30,6 +31,49 @@ func mergedUsrSystem() bool {
 	return false
 }
 
+// Parse modinfo output and return the value of module attributes
+// There may be multiple row with same fieldname so []string
+// is used to return all data.
+func getModData(modname string, fieldname string) []string {
+	out, err := exec.Command("modinfo", modname).Output()
```

Comment:
better to parse module.dep then he command output of modinfo here; which is fundamentally what modinfo does

Reply:
Don't think this is a good idea - the manual _explicitly_ warns against this. From `man modules.dep`

```
       Its text counterpart is located in the same directory with the name modules.dep. The
       text version is maintained only for easy of reading by humans and is in no way used
       by any kmod tool.

       These files are not intended for editing or use by any additional utilities as their
       format is subject to change in the future. You should use the modinfo(8) command to
       obtain information about modules in a future proof and compatible fashion rather than
       touching these files.
```

Reply:
Yes because command line tools are such a great binary interface. Are there no go bindings for libkmod to avoid calling out to other tools? If there aren't at the very least use it sensibly and call `modinfo -F depends <module>`  to get a direct comma seperated list

Reply:
Also you need to run modinfo -k to ensure it picks up the kernel version you're expecting it to be against rather then default to the current version

Reply:
Output format of modinfo -F is not distinct across distro. e.g 
in debian
```
$modinfo -F depends watchdog
name:           watchdog
$modinfo -F depends 9p
fscache,9pnet
$modinfo -F filename 9p
/lib/modules/5.8.0-59-generic/kernel/fs/9p/9p.ko
$modinfo -F filename watchdog
name:           watchdog
(builtin)
$
```
In archbox
```
$modinfo -F depends watchdog
$modinfo -F depends 9p
fscache,9pnet
$modinfo -F filename 9p
/lib/modules/5.12.14-arch1-1/kernel/fs/9p/9p.ko.xz
$modinfo -F filename watchdog
(builtin)
$
```
But the output format of modinfo without flag is expected to be more distinct. So just to be in safe side we are not using any flag to get direct result

Reply:
> Also you need to run modinfo -k to ensure it picks up the kernel version you're expecting it to be against rather then default to the current version

adding -k seems reasonable, will do it

Reply:
> Yes because command line tools are such a great binary interface.

Please don't shoot the messenger.

> Are there no go bindings for libkmod to avoid calling out to other tools?

I've found only a single project, albeit it has been unmaintained for 4 years - so not something I would recommend https://github.com/ElyKar/golang-kmod

Reply:
> Also you need to run modinfo -k to ensure it picks up the kernel version you're expecting it to be against rather then default to the current version

Updated current PR with  "modinfo -k" to get the module information with the correct kernel release.

Reply:
@sjoerdsimons : modinfo -k changes has been applied but ignored modinfo -F suggestion(see reply) in current updated.
Would you like to look again?